src/medicine/medicine_classes.py

```diff
@@ -137,25 +137,37 @@ def append_instance_to_file(self, file_name='instances.p'):
 
         with open(file_name, 'wb') as file:
             pickle.dump(instances_dict, file)
-    
-    
+
+
     def add_medicine(self, amount_of_pills):  # funkcja jeszcze nie gotowa, trzeba ją rozbudować (i to porządnie), tego komentarza nie usuwaj
         """Adds more pills to an instance"""
         assert isinstance(amount_of_pills, int), 'amount_of_pills is supposed to be INT'
         self.current_amount += amount_of_pills
-    
-    
+
+
     def how_much_in_mg(self):
         """Basing on self.current_amount returns amount of the medicine in milligrams"""
         amount_of_pills = self.current_amount
         mg_per_pill = self.dose
         amount_of_mg = amount_of_pills * mg_per_pill
         return amount_of_mg
-    
-    
-    def __str__(self):  # __str__() i __repr__() używam z przyzwyczajenia, tak mi wpojono. W docstringu masz co robią. 
+
+
+    def __str__(self):  # __str__() i __repr__() używam z przyzwyczajenia, tak mi wpojono. W docstringu masz co robią.
         return f'{self.name.title()} medicine with dose {self.dose} mg per pill (daily dose {self.daily_dose} mg) has {self.current_amount} pills left.'
-    
-    
+
+
     def __repr__(self):
         return f'Medicine({self.name}, {self.dose}, {self.daily_dose}, {self.current_amount})'
+
+
+def how_much_in_mg(*medicines):
```

Comment:
testy powinny mieć jaknajmniej kodu - kod żyje w folderze `src`, a testy tylko go wywyołują

Reply:
Czyli testy robię w pliku innym niż src?

Reply:
Teraz widzę, że chyba tak, ale nadal dziwne to jest dla mnie. Ty normalnie żeby sprawdzać funkcjonalność co minutę skaczesz między folderami?

Reply:
`src` jest dla kodu. `tests` jest dla testów. Kod powinien działać bez testów, a najlepiej żeby nie wiedział nawet o istnieniu testów.
Tu trzeba rozróżnić testy jako "piszę se różne rzeczy pomocniczne tworząc kod, ale potem je usunę" od "porządne testy które mają zostać w repozytorium żeby sprawdzić czy kod cały czas jest poprawny". Pierwsza kategoria se rób jak chcesz - to i tak jest efemeryczne. Druga kategoria mieszka w `/tests` i się składa tylko z funkcji `pytest` (+ konfiguracja itp)

Reply:
nie skaczę. Mam w jednym oknie odpalony ipython w którym se sprawdzam jak działają różne funkcje itp. Jak już mi się wydaje że to działa, to robię korekty estetyczne, żeby ładnie wyglądało. Potem się zabieram za testy, które to odpalam w innej konsoli. Testy se piszę i odpalam w koło aż przejdą

Reply:
To Ty programujesz w iPythonie zamiast w VSCode? Czy czegoś nie rozumiem. Sprawdziłem sobie na szybko i nie wygląda na coś czego bym używał do pisania kodu. Czy tam robisz wstępne testy?

Reply:
Ja piszę w [emacs])https://www.spacemacs.org/), bo jestem stary i lepszy od plebsu (Vim też jest do przyjęcia - wszystko inne jest dla fajerów :P). W praktyce to wygląda tak że piszę w emacs, ale mam też 2 (w praktyce częściej z 8) konsoli (w zakładkach, bo nie jestem zwierzem) otwartych. Jedna konsola ma odpalone ipython i tam na bierząco wklejam kod który napisałem który chce sprawdzić czy działa. Wtedy to se wywołuję w tym ipython z różnymi parametrami (bo jak człowiek piszę, więc raczej tylko małe fukncje) i patrzę czy daje sensowne wyniki i czy się nie wykrzacza. A jak już wygląda dobrze, to w emacs otwieram plik z testami do tego co teraz robiłem, piszę testy, a potem w kolejnej konsoli odpalam testy dla tego pliku (albo tylko pojedynczy test).

Jak już mi się wydaję że wszystko gra, to odpalam wszyskie testy (czyli samo `pytest` zamiast np. `pytest tests/test_cokolwiek.py`). Jak wszystko przeszło, to wrzucam to do gita

Reply:
Hardore, to chyba się wydłuża pisanie funkcji trzykrotnie, co? Poza tym - pewnie używasz vima dlatego, że nie zdołałeś z niego wyjść :D 
Będę musiał zrobić sobie jakiś malutki programik żeby sobie przetestować ten Twój system. iPython jest powszechnie używany? 
Jak funkcje łączą się z innymi częściami kodu (choćby używają innych funkcji) to jak robisz testy w terminalu? Przeklejasz cały program?

Reply:
może, ale nie mam pojęcia. Póki co to tylko pracowałem z jednym gościem co szybciej robił ode mnie, i on od początku uzywał emacs'a (ja kiedyś miałem PyCharma), także tego... :D
Ale jest korelacja że najlepsi rzeczywiście używają vim albo emacs. Nie wiem w którą stronę idzie przyczynowość, ale jest.

iPython jest chyba dość powszechnie uzywany - przynajmniej często go widywałem. Python jest językiem interpretowalnym. Czyli każde linie po kolei odpala. Za każdym razem (powiedzmy). To oznacza, że możesz se w dowolnej chwili zmienić dowolną część programu. Ja to robię tak że defakto wklejam cały plik który mnie interesuje (bo jak ma `import ...` to też się odpalą) i wtedy mogę się tym bawić.

Ja jestem trochę dziwny, bo Lisp mnie spaczył. [Tu](https://mikelevins.github.io/posts/2020-12-18-repl-driven/) ktoś to pokrótce tłumaczy różnice. Teraz pewnie za wcześnie, ale bardzo polecam Lispa (w szczególności [Clojure](https://www.braveclojure.com/))

Reply:
Programowanie w vimie brzmi jak wyższy poziom samoumartwiania xd, ale może kiedyś wyjdę z plebsu. A może będę klepał sobie kod i miał wywalone, byle hajs wpadał